src/api/routes/summaries.ts

```diff
@@ -47,23 +49,21 @@ const list = procedure
       transcriptId: z.string(),
       summary: z.string(),
     }))
-  ).query(async ({ input }: { input: SummariesListInput }) => 
-{
-  // TODO: Optimize and use a single query to return final results.
-  const summaries = await Summary.findAll({ 
-    where: { summaryKey: input.key },
-    attributes: ['transcriptId', 'summary'],
+  ).query(async ({ input }: { input: SummariesListInput }) => {
+    return await Summary.findAll({
+      where: {
+        summaryKey: input.key,
+        transcriptId: {
+          [Op.notIn]: Sequelize.literal(`
+            (SELECT "transcriptId" FROM "Summaries" 
```

Comment:
I'd much prefer not to use raw select queries for performance reasons. these queries are very hard to maintain and easy to introduce bugs.

Actually, because we need to have a remodel the summary upload API and it's very likely that this function will need a complete rewrite, let's not worry about this optimization task for now. Sorry about the late notice.